src/openfermion/utils/_hubbard_trotter_error.py

```diff
@@ -0,0 +1,142 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Module to compute Trotter errors in the plane-wave dual basis."""
+from __future__ import absolute_import
+from future.utils import iteritems, itervalues
+
+import numpy
+
+from openfermion.config import *
+from openfermion.hamiltonians import fermi_hubbard
+from openfermion.ops import FermionOperator, normal_ordered
+from openfermion.utils import count_qubits
+from openfermion.utils._dual_basis_trotter_error import (
+    dual_basis_error_bound, dual_basis_error_operator, stagger_with_info)
+
+
+def fermi_hubbard_error_operator(terms, indices=None, is_hopping_operator=None,
+                                 verbose=False):
+    """Determine the difference between the exact generator of unitary
+    evolution and the approximate generator given by the second-order
+    Trotter-Suzuki expansion for the Hubbard model.
+
+    Args:
+        terms: a list of FermionOperators in the Hamiltonian in the
+               order in which they will be simulated.
+        indices: a set of indices the terms act on in the same order as terms.
+        is_hopping_operator: a list of whether each term is a hopping operator.
+        verbose: Whether to print percentage progress.
+
+    Returns:
+        The difference between the true and effective generators of time
+            evolution for a single Trotter step.
+
+    Notes: follows Equation 9 of Poulin et al.'s work in "The Trotter Step
+        Size Required for Accurate Quantum Simulation of Quantum Chemistry".
+    """
+    return dual_basis_error_operator(
+        terms, indices=indices, is_hopping_operator=is_hopping_operator,
+        verbose=verbose)
+
+
+def fermi_hubbard_error_bound(terms, indices=None, is_hopping_operator=None,
+                              verbose=False):
+    """Numerically upper bound the error in the ground state energy
+    for the second-order Trotter-Suzuki expansion.
+
+    Args:
+        terms: a list of single-term FermionOperators in the Hamiltonian
+            to be simulated.
+        indices: a set of indices the terms act on in the same order as terms.
+        is_hopping_operator: a list of whether each term is a hopping operator.
+        verbose: Whether to print percentage progress.
+
+    Returns:
+        A float upper bound on norm of error in the ground state energy.
+
+    Notes:
+        Follows Equation 9 of Poulin et al.'s work in "The Trotter Step
+        Size Required for Accurate Quantum Simulation of Quantum
+        Chemistry" to calculate the error operator.
+    """
+    # Return the 1-norm of the error operator (upper bound on error).
+    return dual_basis_error_bound(
+        terms, indices=indices, is_hopping_operator=is_hopping_operator,
+        verbose=verbose)
+
+
+def simulation_ordered_grouped_hubbard_terms_with_info(
```

Comment:
The name needs to reflect the very specific ordering that is being reflected here. Refer to the arXiv paper in the doc strings so it is clear what the order is.